### PR TITLE
Natural transformation for free freer

### PR DESCRIPTION
**Pull request by xhtmlboi**
For being consistent with `Free_applicative`, `Free_selective` and `Freer_selective`.

### CODE REVIEW

**Comment by gr-im** [comment]
The natural transformation for Freer seems wrong to me. In `Free`, the continuation is provided explicitly in the type, this is not the case in Freer. So how to provide a natural transformation to `Identity` (for example) when the continuation is implicitly defined.

**Review by d-plaindoux — APPROVED** [review]

**Comment by gr-im** [comment]
Could you give an example of "resuming" computation for freer?

File: lib/preface_make/free_monad.ml [hunk 1]
```diff
@@ -7,6 +7,16 @@ module Via_map (F : Preface_specs.Functor.CORE) = struct
 
   let perform f = Bind (F.map (fun a -> Return a) f)
 
+  module To_monad (Monad : Preface_specs.Monad.CORE) = struct
+    type natural_transformation = { transform : 'a. 'a f -> 'a Monad.t }
+
+    let rec run transformation = function
+      | Return a -> Monad.return a
+      | Bind g ->
+        Monad.bind (fun x -> run transformation x) (transformation.transform g)
```
[the remark this answers is not in the document]
Comment (xhtmlboi) [reply lib/preface_make/free_monad.ml]: Done! (No particular reason, thanks)

**Comment by xhtmlboi** [comment]
> The natural transformation for Freer seems wrong to me. In `Free`, the continuation is provided explicitly in the type, this is not the case in Freer. So how to provide a natural transformation to `Identity` (for example) when the continuation is implicitly defined.

You were absolutely right, I modified the transformation on the Freer and implemented your example in test. Thank you for that! I was probably a bit fast in my implementation :)

**Review by gr-im — APPROVED** [review]
Thanks, It is better (IMHO)

**Review by xvw — APPROVED** [review]

**Review by d-plaindoux — APPROVED** [review]

**Pull request merged**